src/backends/pytorch/libtorch_backend.cc

```diff
@@ -460,324 +546,226 @@ LibTorchBackend::Context::GetOutputTensor(
-    return Status(
-        Status::Code::INTERNAL,
-        "dynamic batch size " + std::to_string(total_batch_size) + " for '" +
-            name_ + "', max allowed is " + std::to_string(max_batch_size_));
+    InferenceRequest::RespondIfError(
+        requests,
+        Status(
+            Status::Code::INTERNAL,
+            "dynamic batch size " + std::to_string(total_batch_size) +
+                " for '" + name_ + "', max allowed is " +
+                std::to_string(max_batch_size_)),
+        true /* release_requests */);
+    return;
+  }
+
+  // At this point we are committed to running inference with all
+  // 'requests'. Create a response for each request. During input
+  // processing if there is an error with any request that error will
+  // be sent immediately with the corresponding response (and the
+  // response unique_ptr will then be nullptr). The request object
+  // itself will not be released until after all inferencing is done
+  // (below) as we may need to access the request object when
+  // determine how to process outputs (for example, even if we don't
+  // need the outputs for a request that has an error, we do need to
+  // know the size of those outputs associated with the request so we
+  // can skip them in the output tensors).
+  std::vector<std::unique_ptr<InferenceResponse>> responses;
+  responses.reserve(requests.size());
+
+  for (auto& request : requests) {
+    std::unique_ptr<InferenceResponse> response;
+    Status status = request->ResponseFactory().CreateResponse(&response);
+    if (!status.IsOk()) {
+      InferenceRequest::RespondIfError(request, status);
+      response.reset();
+    }
+
+    responses.emplace_back(std::move(response));
   }
 
   size_t input_count = repr_input_request->ImmutableInputs().size();
 
-  // Hold reference to each buffer of input data so that it stays
+  // Hold reference to each buffer of input data to that it stays
   // until the inference has completed.
-  std::vector<InputMetaData> input_meta_data(input_count);
+  std::vector<std::unique_ptr<AllocatedMemory>> input_buffers;
 
   // Store input and output tensors
   std::vector<torch::jit::IValue> inputs_(input_count);
   std::vector<torch::Tensor> outputs_;
-  std::vector<InputInfo> inputs;
 
-  // Collect input metadata. FIXME override inputs from controls
-  // should be known from the model configuration at load time and so
-  // they should be processed then to initialze
-  // input_index_map_. Since they are not we do it here for every
-  // request which is unnecessary perf overhead.
-  bool cuda_copy = false;
-  for (const auto& pr : repr_input_request->ImmutableInputs()) {
-    const InferenceRequest::Input* input = pr.second;
-    const std::string& name = input->Name();
-    int ip_index;
+  // Collect the request inputs into contiguous input tensors.
+  // FIXME override inputs from controls should be known from the model
```

Comment:
Is this comment still true?

Reply:
No. Validating Control Inputs during model loading now